project/src/main/webapp/document.jsp

```diff
@@ -177,6 +187,173 @@
         a.click();
       }
 
+      //Create comment
+      function comment() {
+        var date = new Intl.DateTimeFormat('en-US', {
+          month: 'long',
+          day: 'numeric',
+          hour: 'numeric',
+          minute: 'numeric'
+        }).format(new Date());
+  
+        var startPos = codeMirror.getCursor(true);
+        var endPos = codeMirror.getCursor(false);
+        endPos.ch += 1;
+
+        for(var i = 0; i < codeMirror.getLine(startPos.line).length; i++)  {
+          if(codeMirror.getLine(startPos.line).charCodeAt(i) > 255) {
+            //deal with multiple comments on same line
+            return;
+          }
+        }
+
+        for(var i = 0; i < codeMirror.getLine(endPos.line).length; i++)  {
+          if(codeMirror.getLine(endPos.line).charCodeAt(i) > 255) {
+            //deal with multiple comments on same line
+            return;
+          }
+        }
+
+        // Create comment getElementsByName
+        document.getElementById('comment-container').innerHTML += '<comment-component name="<%= user.getNickname() %>" hash="<%= (String)request.getAttribute("documentHash") %>" date="' + date + '"></comment-component>';
+        document.querySelector('comment-component').firepad = firepad;
+        document.querySelector('comment-component').codeMirror = codeMirror;
+      }
+
+      // When a comment is submitted, pass the data to the Comment servlet
+      function subComment() {
```

Comment:
rename to submitComment, which is more descriptive

Reply:
will do

Reply:
+1